### PR TITLE
GitLab package for DC/OS

### PR DESCRIPTION
**Pull request by ssk2**
- Use BRIDGE network
- Add HAProxy vitualhost config, use marathon-lp
- Add support for external Postgres and Redis configuration
- Add ssh-port config option, and set a ssh service port in the port mapping
- Prevent scaling beyond one app if not highly available.
- Add external volumes configuration.
- Add smtp email config to the template
- Includes option to install the the EE
- Add configurable persistent volumes.

cc @twk3

### CODE REVIEW

**Comment by ssk2** [comment]
Still outstanding:
- [ ] Successful index file creation
- [ ] Add minimumVersion for DC/OS 1.8

File: repo/packages/G/gitlab/0/config.json [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "single-node": {
+      "description": "Configuration related to running a single instance version of GitLab. If high availability is enabled, these settings will be ignored.",
+      "type": "object",
+      "properties": {
+        "persistent-volumes": {
```
Comment (jdef) [inline repo/packages/G/gitlab/0/config.json]: nit: this name is confusing. suggest `local-volumes`

Comment (ssk2) [inline repo/packages/G/gitlab/0/config.json]: Good idea.

**Comment by KenanSulayman** [comment]
Retest this please

**Comment by ssk2** [comment]
@KenanSulayman we've got a couple of things to test before we can merge!

**Comment by KenanSulayman** [comment]
I know, I know.. obviously that's why it's not merged, yet. But I'm that impatient kind of person 🍻

**Comment by tobilg** [comment]
Has this yet landed in DC/OS 1.8? I thought I read an announcement: https://twitter.com/mesosphere/status/770766837734739968

**Comment by ssk2** [comment]
It'll be merged in when DC/OS 1.8 is made generally available in a
fortnight. (That announcement was a webinar previewing it :))

**Pull request merged**